scene/animation/animation_mixer.cpp

```diff
@@ -1068,11 +1069,13 @@ void AnimationMixer::blend_capture(double p_delta) {
 }
 
 void AnimationMixer::_blend_calc_total_weight() {
+	thread_local BitSet processed_indices;
 	for (const AnimationInstance &ai : animation_instances) {
 		Ref<Animation> a = ai.animation_data.animation;
 		real_t weight = ai.playback_info.weight;
 		Vector<real_t> track_weights = ai.playback_info.track_weights;
-		Vector<int> processed_indices;
+		processed_indices.clear();
+		processed_indices.resize(track_count);
```

Comment:
Is there a maximum track count? If so and this is called often, you can also allocate on the stack with `alloca`. This is done in a lot of performance sensitive cases. (I haven't profiled this.)

Reply:
clear() and resize() won't actually affect the capacity of the vector in most cases. What it's simply doing here is just setting the size to 0, and then the resize sets the new size and clears all bits to its default (false), which is why these two steps are joined together.

The only allocations that happen are when across multiple calls (in the same thread), track_count increases enough to justify increasing the capacity. I believe it's either through power of twos or by 1.5 growth ratios. I've talked with @RandomShaper in the past about alloca, but it comes with the drawback of potentially introducing a bit of a scalability problem if the function ever needs a very large amount of data and stack memory is very limited.

Reply:
Yes, and anyway we can apply an enhancement later for all the `thread_local` vectors such that they are backed by memory from an arena allocator or something, getting the best of both worlds.